### ISSUE TITLE
Magento Tax Rules need order amount thresholds

### ISSUE THREAD

**Issue by gwharton** [magento/magento2#31746]
### Description (*)
It should be possible to state an order amount threshold for each tax rule.

### Expected behavior (*)
You should be able to specify a minimum order value for the tax rule, so the tax is only applied when the order amount is above the threshold.

You should be able to specify a maximum order value for the tax rule, so the tax is only applied when the order amount is below the threshold.

### Benefits
This is required for the worldwide sellers selling to the UK from 1st January 2021 and for worldwide sellers selling to the EU from 1st July 2021. Tax should only be collected on orders below 135GBP or 150EUR i believe it is.

**Comment by YiffyToys** [comment]
Closed the community-features#317 I created as it's a duplicate of this issues/31746.

What happens after "ready for grooming"? Brexit was 1 month ago, so people ARE loosing money
by refusing any orders <135GBP already.

(If "community-features" is the right place to handle this, we should reopen that and close this.)

**Comment by hostep** [comment]
I've been stalking some people on Magento's Slack (@maghamed & @sidolov) to make this a super high priority, unfortunately nobody can or wants to tell me the status at this point...

@gwharton is also currently implementing a feature similar to this himself for his own needs, he might share what he has but has no obligations to do so.

**Comment by gwharton** [comment]
In the next couple of days I'll be upping a module to github with a view to getting some help collaborating and testing a module that can serve as an interim solution until Magento adopt/implement their own long term solution.

**Comment by gwharton** [comment]
https://github.com/gwharton/autocustomergroup

**Comment by gwharton** [comment]
Be Gentle please. Issues/PR's welcome. If you want to collaborate, I'd be more than happy to open the repo to you if you think you can improve it. I'm not expert on either PHP or magento, so if anything looks dodgy, feel free to point it out, or improve it yourself and feed the changes back.

**Comment by YiffyToys** [comment]
Dynamically changing the customer group for each order of that customer. Sounds like a temporary workaround that could actually work until this ticket gets assigned and worked on to create a proper solution.

**Comment by YiffyToys** [comment]
Still needed.
Brexit is still a thing.

**Comment by hostep** [comment]
@sidolov: what would be status here?

Brexit happened 4 months ago and the 1st of July is not that far away anymore ...

**Comment by YiffyToys** [comment]
We just had to refuse 3 more orders <135GBP because we can't set up the correct tax rules.

**Comment by gwharton** [comment]
Have you tried my module https://github.com/gwharton/autocustomergroup it's now on packagist too. It is beta, but better than nothing and seems fairly stable.

**Comment by YiffyToys** [comment]
Not yet. We've been busy fixing interoperability issues between Amazon and DHL plugins recently.

**Comment by gwharton** [comment]
Quick note to Magento core Devs. If you want to use any of the code from the module I have released, feel free to reach out. I'm more than happy to sign over the whole thing to Adobe for nothing.

**Comment by YiffyToys** [comment]
> Closed https://github.com/magento/magento2/issues/31746 as not planned
WTF @gwharton ?

Why was this closed?
BREXIT and HMRC tax-rules on VAT are still there.
For many years now we are just unable to allow British customers to place orders below 135£ because Magento can not handle the UK tax laws.

The module assigning customer groups is just a band-aid as it doesn't work if the customer chooses a different delivery address (e.g. in Northern Ireland or France) or pick-up without shipping.

**Comment by gwharton** [comment]
Sorry Yiffy. I'm just a retailer like you who wants his store to work properly, but i'm rapidly losing faith in Adobe's ability to fix anything recently. I've had issues open for years and years and they just get ignored and ignored. I'm getting sick of participating in Magento development. 

I agree the auto customer group module is a band aid. I wrote it because I needed it for my store. If you'd like to see it include other scenareos, you're welcome to contribute.

**Comment by YiffyToys** [comment]
I'd prefer an order amount thresholds criteria (GB) and a zip-code criteria (Northern Ireland exception) to be possible in Magento2 itself.
That's why I don't understand why this ticket is being closed to never be implemented in Magento2.

I don't understand enough about of the internals of Magento, it's developer/code documentation, unit tests and logging to write a good and well tested patch including everything outside the rule that needs updating myself at this point.
Since my shop currently isn't able to comply with EU and UK invoice format at the same time, I may see about a patch to at least add a zip-code like there's a country criteria.
But not until after the next big trade event in a month. That currently has priority.

**Comment by gwharton** [comment]
I'm not part of Magento/Adobe. I'm just sick of having this open issue laying around for years on end and getting ignored. It wasn't a statement from Magento to say they won't do anything.

I'm sure internally at Magento, it's on someone's long list of things that need doing, but are too hard, or that aren't understood enough.

Unless you are a developer yourself, or you can afford to pay a team of developers, Adobe Commerce, out of the box, is just not shop ready IMHO.

If you'd like to give a bit more detail on the limitations of my autocustomergroup module, i'll try to get some time to take a look at them and scope them out, as I wouldn't hold your breath for anything from Magento on this.

**Comment by twosg** [comment]
Hello togehter,
Although this is going to be a little advertisement I wanted to mention that my extension supports calculating the 135 GBP threshold (with currency conversion) for shipping to UK as well as "the opposite" of 150 EUR when shipping into the EU.
That is all possible without any customer group change with the dynamic customer tax class feature.
[EU VAT Enhanced for Magento 2](https://geissweb.com/eu-vat-enhanced-for-magento-2.html)
Just mentioning it as a solution for this issue.
